source/common/network/connection_impl.cc

```diff
@@ -302,6 +297,11 @@ void ConnectionImpl::readDisable(bool disable) {
     ASSERT(read_enabled_);
     read_enabled_ = false;
 
+    if (state() != State::Open || file_event_ == nullptr) {
+      // If readDisable is called on a closed connection, do not crash.
+      return;
```

Comment:
the bug fix is to allow read_disable_count_ to be adjusted prior to the exit?

Reply:
The fix is to allow read_enabled_ to transition to true in the branch below.